includes/class-wcs-att-cart.php

```diff
@@ -269,36 +290,63 @@ public static function apply_subscription_schemes( $cart ) {
 
 			if ( isset( $cart_item[ 'wcsatt_data' ] ) ) {
 
-				// If subscription schemes are available at cart-level, set them on the product object.
-				if ( ! empty( $cart_level_schemes ) ) {
-					WCS_ATT_Product_Schemes::set_subscription_schemes( $cart_item[ 'data' ], $cart_level_schemes );
-				}
+				/*
+				 * If renewing a subscription, delete all scheme-releted data from the object:
+				 * WCS has already linked it with a subscription order and will even copy the right total from there.
+				 */
+				if ( isset( $cart_item[ 'subscription_renewal' ] ) ) {
 
-				// Initialize subscription scheme data.
-				$cart_item[ 'wcsatt_data' ][ 'active_subscription_scheme' ] = self::get_subscription_scheme_to_apply( $cart_item );
+					WCS_ATT_Product_Schemes::set_subscription_schemes( $cart->cart_contents[ $cart_item_key ][ 'data' ], array() );
```

Comment:
Note there's an *s*, here we are removing all schemes on the object. This is because renewal cart items don't require any scheme data, as everything is copied over from the subscription object and handled by WCS.